shell/src/chain_manager.rs

```diff
@@ -631,6 +633,38 @@ impl ChainManager {
                             }.into(),
                             topic: ShellChannelTopic::ShellEvents.into(),
                         }, Some(ctx.myself().into()));
+
+                    // handle operations
+                    if inject_data.block_header.level() > 1 {
```

Comment:
@adonagy 
I see that inject_data.block_header is cloned upper to `block_header_with_hash`,
can `block_header_with_hash` be used here also?
because  later you do the same clone: `let header = inject_data.block_header.clone();` which is the same as `block_header_with_hash`

Reply:
Indeed, I'll reuse block_header_with_hash.